go/client/cmd_chat_api.go

```diff
@@ -224,24 +224,29 @@ func (c *CmdChatAPI) ReadV1(ctx context.Context, opts readOptionsV1) Reply {
 	var thread Thread
 	thread.Messages = make([]MsgSummary, len(threadView.Messages))
 	for i, m := range threadView.Messages {
-		thread.Messages[i] = MsgSummary{
-			ID: m.ServerHeader.MessageID,
-			Channel: ChatChannel{
-				Name:      m.MessagePlaintext.ClientHeader.TlfName,
-				TopicType: strings.ToLower(m.MessagePlaintext.ClientHeader.Conv.TopicType.String()),
-			},
-			Sender: MsgSender{
-				UID:      m.MessagePlaintext.ClientHeader.Sender.String(),
-				DeviceID: m.MessagePlaintext.ClientHeader.SenderDevice.String(),
-			},
-			SentAt:   int64(m.ServerHeader.Ctime / 1000),
-			SentAtMs: int64(m.ServerHeader.Ctime),
-		}
-		if len(m.MessagePlaintext.MessageBodies) > 0 {
-			thread.Messages[i].Content = c.convertMsgBody(m.MessagePlaintext.MessageBodies[0])
+		version, err := m.MessagePlaintext.Version()
+		if err != nil {
+			return c.errReply(err)
 		}
-		if len(m.MessagePlaintext.MessageBodies) > 1 {
-			c.G().Log.Warning("message %v had multiple bodies", m.ServerHeader.MessageID)
+		switch version {
+		case keybase1.MessagePlaintextVersion_V1:
+			v1 := m.MessagePlaintext.V1()
+			thread.Messages[i] = MsgSummary{
+				ID: m.ServerHeader.MessageID,
+				Channel: ChatChannel{
+					Name:      v1.ClientHeader.TlfName,
+					TopicType: strings.ToLower(v1.ClientHeader.Conv.TopicType.String()),
+				},
+				Sender: MsgSender{
+					UID:      v1.ClientHeader.Sender.String(),
+					DeviceID: v1.ClientHeader.SenderDevice.String(),
+				},
+				SentAt:   int64(m.ServerHeader.Ctime / 1000),
```

Comment:
I realize this has been inherited from previous code, but we should really never do these manipulations by hand, we should use Go types to make sure we're doing this stuff correctly.